### PR TITLE
[python] added f-string to python-package/lightgbm/basic.py

### PR DESCRIPTION
**Pull request by NovusEdge**

### CODE REVIEW

**Comment by NovusEdge** [comment]
#4136

**Comment by StrikerRUS** [comment]
Please update
https://github.com/microsoft/LightGBM/blob/893702204563376dff1a20c58efcc42c8181b279/python-package/lightgbm/basic.py#L537-L539
https://github.com/microsoft/LightGBM/blob/893702204563376dff1a20c58efcc42c8181b279/python-package/lightgbm/basic.py#L2426
https://github.com/microsoft/LightGBM/blob/893702204563376dff1a20c58efcc42c8181b279/python-package/lightgbm/basic.py#L2431

**Comment by jameslamb** [comment]
> Please update

updated in https://github.com/microsoft/LightGBM/pull/4143/commits/a2cfecbc1d379f50e89b6a4fb7e33b97d8df3346 and merged in latest `master`

File: python-package/lightgbm/basic.py [hunk 1]
```diff
@@ -1774,8 +1768,7 @@ def set_feature_name(self, feature_name):
             self.feature_name = feature_name
         if self.handle is not None and feature_name is not None and feature_name != 'auto':
             if len(feature_name) != self.num_feature():
-                raise ValueError("Length of feature_name({}) and num_feature({}) don't match"
-                                 .format(len(feature_name), self.num_feature()))
+                raise ValueError(f"Length of feature_name({len(feature_name)}) and num_feature({self.num_feature}) don't match")
```
Comment (StrikerRUS) [inline python-package/lightgbm/basic.py:1771]:
```suggestion
                raise ValueError(f"Length of feature_name({len(feature_name)}) and num_feature({self.num_feature()}) don't match")
```

File: python-package/lightgbm/basic.py [hunk 2]
```diff
@@ -3272,9 +3260,7 @@ def feature_name(self):
             raise ValueError("Length of feature names doesn't equal with num_feature")
         if reserved_string_buffer_size < required_string_buffer_size.value:
             raise BufferError(
-                "Allocated feature name buffer size ({}) was inferior to the needed size ({})."
-                .format(reserved_string_buffer_size, required_string_buffer_size.value)
-            )
+                f"Allocated feature name buffer size ({reserved_string_buffer_size}) was inferior to the needed size ({required_string_buffer_size.value}.")
```
Comment (StrikerRUS) [inline python-package/lightgbm/basic.py:3263]:
```suggestion
                f"Allocated feature name buffer size ({reserved_string_buffer_size}) was inferior to the needed size ({required_string_buffer_size.value}).")
```

File: python-package/lightgbm/basic.py [hunk 3]
```diff
@@ -2423,12 +2414,12 @@ def create_node_record(tree, node_depth=1, tree_index=None,
                                feature_names=None, parent_node=None):
 
             def _get_node_index(tree, tree_index):
-                tree_num = str(tree_index) + '-' if tree_index is not None else ''
+                tree_num = f'{tree_index}-' if tree_index is not None else ''
                 is_split = _is_split_node(tree)
                 node_type = 'S' if is_split else 'L'
                 # if a single node tree it won't have `leaf_index` so return 0
                 node_num = str(tree.get('split_index' if is_split else 'leaf_index', 0))
```
Comment (StrikerRUS) [inline python-package/lightgbm/basic.py:2421]:
```suggestion
                node_num = tree.get('split_index' if is_split else 'leaf_index', 0)
```

**Review by StrikerRUS — CHANGES_REQUESTED** [review]
Thanks! Just few suggestions.

**Review by StrikerRUS — APPROVED** [review]

**Pull request merged**